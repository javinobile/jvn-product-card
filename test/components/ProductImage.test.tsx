import React from 'react';
import renderer from 'react-test-renderer'
import { ProductCard } from "../../src/components/ProductCard";
import { product, product2 } from "../data/products";
import { ProductImage } from '../../src/components/ProductImage';

describe('ProductImage', () => {
    test('Debe de mostrar el componente sin imagen', () => {

        const wrapper = renderer.create(
            <ProductImage img='http://hola.jpg'  />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

    test('Debe de mostrar el componente con imagen', () => {
        const wrapper = renderer.create(
            <ProductCard product={product}>
                {
                    () => (
                        <ProductImage img={product2.img}  />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});