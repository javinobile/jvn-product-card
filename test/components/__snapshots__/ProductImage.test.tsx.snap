// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente con imagen 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Debe de mostrar el componente sin imagen 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="http://hola.jpg"
/>
`;
